File: AspectJAOP/app/src/main/java/com/cyq/aspectjaop/MainActivity.java
package com.cyq.aspectjaop;

import android.content.Intent;
import android.media.MediaPlayer;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.cyq.aspectjaop.annotation.ClickBehavior;
import com.cyq.aspectjaop.annotation.LoginCheck;

public class MainActivity extends AppCompatActivity {

    private final static String TAG = "netease >>> ";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    // 登录点击事件（用户行为统计）
    @ClickBehavior("登录")
    public void login(View view) {
        Log.e(TAG, "模拟接口请求……验证通过，登录成功！");
    }

    // 用户行为统计
    @ClickBehavior("我的专区")
    @LoginCheck
    public void area(View view) {
        Log.e(TAG, "开始跳转到 -> 我的专区 Activity");
        startActivity(new Intent(this, OtherActivity.class));
    }

    // 用户行为统计
    @ClickBehavior("我的优惠券")
    @LoginCheck
    public void coupon(View view) {
        Log.e(TAG, "开始跳转到 -> 我的优惠券 Activity");
        startActivity(new Intent(this, OtherActivity.class));
    }

    // 用户行为统计
    @ClickBehavior("我的积分")
    @LoginCheck
    public void score(View view) {
        Log.e(TAG, "开始跳转到 -> 我的积分 Activity");
        startActivity(new Intent(this, OtherActivity.class));
    }

}

File: JetpackDemo/app/src/main/java/com/cyq/jetpack/MainActivity.kt
package com.cyq.jetpack

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.os.Message
import android.util.Log
import android.view.View
import com.cyq.jetpack.navigation.NavigationActivity
import com.cyq.jetpack.paging.PageKeyedActivity
import com.cyq.jetpack.paging.PositionalPagingActivity
import kotlinx.android.synthetic.main.activity_main.*
import kotlin.concurrent.thread

class MainActivity : AppCompatActivity(), View.OnClickListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnDataBinding.setOnClickListener(this)
        btnPaging.setOnClickListener(this)
        btnPaging2.setOnClickListener(this)
        btnNavigation.setOnClickListener(this)

        val msg = Message()
        msg.what = 101
        Log.e("test","begin:${System.currentTimeMillis()}")
        handler.sendMessageDelayed(msg, 5000)
        thread {
            Thread.sleep(1000)
            handler.obtainMessage()
        }
        btnThread.setOnClickListener(this)
    }

   private val handler = Handler(object : Handler.Callback {
        override fun handleMessage(msg: Message): Boolean {
            Log.e("test","end:${System.currentTimeMillis()}")
            Log.e("test", "---handler---")
            return true
        }
    })

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.btnDataBinding ->
                startActivity(Intent(this, DataBindingActivity::class.java))
            R.id.btnPaging ->
                startActivity(Intent(this, PositionalPagingActivity::class.java))
            R.id.btnPaging2 ->
                startActivity(Intent(this, PageKeyedActivity::class.java))
            R.id.btnThread ->
                startActivity(Intent(this, ThreadActivity::class.java))
            R.id.btnNavigation ->
                startActivity(Intent(this,NavigationActivity::class.java))
        }
    }
}